README
This set of SlackBuilds should get Bumblebee up and running on a
Slackware based NVIDIA Optimus setup.


Slackware Docs:
  A wiki page on the Slackware docs has been put to gether by another
  slacker, TommyC. I recommend you follow the wiki page for a more
  in-depth guide.

  http://docs.slackware.com/howtos:hardware:nvidia_optimus


HowTo:
  If you have never used a SlackBuild before, please refer to the HowTo
  on SlackBuilds.org: http://slackbuilds.org/howto/

  All the Slackbuild scripts were designed to be run as root, with root's
  environment.

Note:
  Several of these SlackBuilds support a COMPAT32 option which
  allows 32-bit binaries to be built and packaged.  This does
  require that the system is multilib, otherwise the SlackBuilds 
  will fail.

  For more information on slackware multilib, visit AlienBOB's wiki:
  http://alien.slackbook.org/dokuwiki/doku.php?id=slackware:multilib


1.  Download the sources:
      $ ./download.sh

2.  Create group: bumblebee:
      # su -
      # groupadd bumblebee

    Add users to the group:
      # usermod -G bumblebee -a USERNAME

    Note: you will need to re-login as the user for this to take effect.

3.  Build and install: libbsd

    # cd libbsd
    # ./libbsd.Slackbuild
    # upgradepkg --install-new /tmp/libbsd-<ver-arch-build>_bbsb.txz
    # cd ..

4.  Build and install: bumblebee

    # cd bumblebee
    # ./bumblebee.Slackbuild
    # upgradepkg --install-new /tmp/bumblebee-<ver-arch-build>_bbsb.txz
    # cd ..

5.  Build and install: bbswitch (Optional)

    This in an optional requirement.  This is the kernel module that allows 
    the Nvidia card to be turned off, potentially saving you power.  If you 
    do not need power management or the ability to turn off the nVidia chip, 
    you can skip this.

    # cd bbswitch
    # ./bbswitch.Slackbuild
    # upgradepkg --install-new /tmp/bbswitch-<ver-arch-build>_bbsb.txz
    # cd ..

6.  Build and install: libvdpau (Optional, not needed if using nouveau)

    # cd libvdpau
    # ./libvdpau.Slackbuild
    # upgradepkg --install-new /tmp/libvdpau-<ver-arch-build>_bbsb.txz
    # cd ..

7.  Build and install: nvidia-kernel (Optional, not needed if using nouveau)
    
    # cd nvidia-kernel
    
    For pure 32 or 64 bit systems, build via:
    # ./nvidia-kernel.Slackbuild


    If the system is x86_64 based, 32-bit compatible binaries and
    libraries can be built via:
    # COMPAT32=yes ./nvidia-kernel.SlackBuild

    Then install:
    # upgradepkg --install-new /tmp/nvidia-kernel-<ver-arch-build>_bbsb.txz
    # cd ..

8.  Build and install: nvidia-bumblebee (Optional, not needed if using nouveau)

    # cd nvidia-bumblebee
    
    For pure 32 or 64 bit systems, build via:
    # ./nvidia-bumblebee.Slackbuild


    If the system is x86_64 based, 32-bit compatible binaries and
    libraries can be built via:
    # COMPAT32=yes ./nvidia-bumblebee.SlackBuild

    Then install:
    # upgradepkg --install-new /tmp/nvidia-bumblebee-<ver-arch-build>_bbsb.txz
    # cd ..

9.  Build and install primus

    # cd primus
    
    For pure 32 or 64 bit systems, build via:
    # ./primus.Slackbuild


    If the system is x86_64 based, 32-bit compatible binaries and
    libraries can be built via:
    # COMPAT32=yes ./primus.SlackBuild

    Then install:
    # upgradepkg --install-new /tmp/primus-<ver-arch-build>_bbsb.txz
    # cd ..

    primus speeds can be much improved by running:
      vblank_mode=0 primusrun

10. Run the rc.bumblebee script:
     # chmod +x /etc/rc.d/rc.bumblebeed
     # /etc/rc.d/rc.bumblebeed start

    If you'd like to have bumblebee autostart with the system, you will
    need to add the following lines to: /etc/rc.d/rc.local:

    if [ -x /etc/rc.d/rc.bumblebeed ]; then
      /etc/rc.d/rc.bumblebeed start
    fi

    You can also go a step further by having bumblebeed stop with your
    system by adding the following lines to: /etc/rc.d/rc.local_shutdown:

    if [ -x /etc/rc.d/rc.bumblebeed ]; then
      /etc/rc.d/rc.bumblebeed stop
    fi

11. Now an application can run with optirun:
    
    $ vblank_mode=0 primusrun glxgears


ChangeLog.txt
+--------------------------+
Mon Nov 01 15:53:32 CET 2021
Upgraded to nvidia driver 470.82.00

+--------------------------+
Thu Aug 19 15:53:32 CET 2021
Upgraded to nvidia driver 470.63.01

+--------------------------+
Sat Jun 12 15:53:32 CET 2021
Upgraded to nvidia driver 460.84

+--------------------------+
Sat May 15 15:53:32 CET 2021
Upgraded to nvidia driver 460.80

+--------------------------+
Ven Mar 26 15:53:32 CET 2021
primus_vk upload vk_layer_dispatch_table.h patched

+--------------------------+
Sat Feb 27 15:53:32 CET 2021
Upgraded to nvidia driver 460.56

+--------------------------+
Thu Jan 26 15:53:32 CET 2021
Upgraded to primus_vk 1.6.1

+--------------------------+
Thu Jan 21 15:53:32 CET 2021
Upgraded to nvidia driver 460.32.03
bumblebee patch for gcc10

+--------------------------+
Sun Dec 20 15:53:32 CET 2020
Upgraded to nvidia driver 455.45.01
bbswitch patch for kernel 5.6+

+--------------------------+
Sun Oct 11 15:53:32 CET 2020
Upgraded to nvidia driver 450.80.02

+--------------------------+
Sun Aug 30 15:53:32 CET 2020
Upgraded to nvidia driver 450.66

+--------------------------+
Sun Jul 12 15:53:32 CET 2020
Upgraded to nvidia driver 450.57

+--------------------------+
Thu Jun 25 12:31:32 CET 2020
Upgraded to nvidia driver 440.100

+--------------------------+
Sun Apr 12 11:41:40 CET 2020
Upgraded to nvidia driver 440.82

+--------------------------+
Sat Mar  7 09:41:40 CET 2020
Upgraded to nvidia driver 440.64

+--------------------------+
Sat Jan 13 22:52:12 UTC 2018
nvidia-bumblebee: upgraded.
nvidia-kernel: upgraded.
libbsd: upgraded

+--------------------------+
Sat Mar 25 22:58:12 UTC 2017
nvidia-bumblebee: upgraded.
nvidia-kernel: upgraded.

+--------------------------+
Fri Aug 9 15:21:32 UTC 2013
libbsd: upgraded
SLACKBUILDS.TXT: removed

+--------------------------+
Wed Aug 7 00:36:48 UTC 2013
All Slackbuild scripts now auto version set. This should help future updates.
libvdpau: upgraded
primus: upgraded

+--------------------------+
Tue Aug 06 12:49:16 UTC 2013
nvidia drivers updated to support 3.10 kernel series
nvidia-bumblebee: upgraded
nvidia-kernel: upgraded

+--------------------------+
Tue Jul 02 16:55:45 UTC 2013
Progress by deletion.  Primus has effectively replaced VirtualGL
and is a great improvement in speed and stability when used with
bumblebee. Also, the mesa package in -current now has the library
required for primus.
libjpeg-turbo: removed.
mesa: removed.
VirtualGL: removed.

+--------------------------+
Fri Jun 28 13:30:15 UTC 2013
The Nvidia drivers are now upgraded to the latest 319.32 release.
README: updated.
SLACKBUILDS.TXT: updated.
nvidia-bumblebee: upgraded.
nvidia-kernel: upgraded.


crazybee.sh
#!/bin/sh
## run as root

# Ryan P. C. McQuen | Everett, WA | ryanpcmcquen@member.fsf.org

# curl https://raw.githubusercontent.com/ryanpcmcquen/linuxTweaks/master/slackware/crazybee.sh | sh

STABLE=${STABLE:-no}
REINSTALL=${REINSTALL:-no}
UPDATE=${UPDATE:-yes}

if [ ! $UID = 0 ]; then
  cat << EOF
This script must be run as root.
EOF
  exit 1
fi

if [ -e ~/Bumblebee-SlackBuilds/crazybee.sh ]; then
  echo "Reinstall!"
  export REINSTALL=${REINSTALL:-yes}
elif [ -d ~/Bumblebee-SlackBuilds/ ]; then
  cat << EOF
You already have the bumblebee directory,
please rename to avoid changes you may have made.
EOF
  exit 1
fi

if [ "$(uname -m)" = "x86_64" ] && [ -e /lib/libc.so.6 ]; then
  export COMPAT32=yes;
fi

install_latest_pkg() {
  PACKAGE=$1
  if [ -d "$PACKAGE/" ]; then
    cd $PACKAGE/
  else
    cd ../$PACKAGE/
  fi
  sh $PACKAGE.SlackBuild
  ls -t --color=never /tmp/$PACKAGE-*_bbsb.txz | head -1 | xargs -i upgradepkg --reinstall --install-new {}
}

install_latest_pkg_compat() {
  PACKAGE=$1
  if [ -d "$PACKAGE/" ]; then
    cd $PACKAGE/
  else
    cd ../$PACKAGE/
  fi
  if [ "$COMPAT32" = yes ]; then
    COMPAT32=yes sh $PACKAGE.SlackBuild
  else
    sh $PACKAGE.SlackBuild
  fi
  ls -t --color=never /tmp/$PACKAGE-*_bbsb.txz | head -1 | xargs -i upgradepkg --reinstall --install-new {}
}

if [ "$UPDATE" = "yes" -a "$REINSTALL" = "no" ]; then
  cd
  git clone https://github.com/WhiteWolf1776/Bumblebee-SlackBuilds.git
  cd Bumblebee-SlackBuilds/
fi

if [ "$UPDATE" = "yes" -a "$STABLE" = "yes" ]; then
  git checkout 14.1
fi

## in case there are updates ;-)
if [ "$UPDATE" = "yes" -a "$REINSTALL" = "yes" ]; then
  git pull
fi

## let's get some tarballs!
sh download.sh

groupadd bumblebee
## add all non-root users (except ftp) to bumblebee group
getent passwd | grep "/home" | cut -d: -f1 | sed '/ftp/d' | xargs -i usermod -G bumblebee -a {}

install_latest_pkg libbsd

install_latest_pkg bumblebee

install_latest_pkg bbswitch

install_latest_pkg_compat primus
install_latest_pkg_compat primus_vk

cd ../nouveau-blacklist/
ls -t --color=never xf86-video-nouveau-blacklist-*noarch-[0-9]*.txz | \
    head -1 | xargs -i upgradepkg --install-new --reinstall {}
if [ -z "$(grep xf86-video-nouveau /etc/slackpkg/blacklist)" ]; then
  echo xf86-video-nouveau >> /etc/slackpkg/blacklist
fi
if [ -z "$(grep _bbsb /etc/slackpkg/blacklist)" ]; then
  echo "[0-9]+_bbsb" >> /etc/slackpkg/blacklist
fi

if [ "$STABLE" = "yes" ]; then
  install_latest_pkg libvdpau
fi

install_latest_pkg nvidia-kernel

install_latest_pkg_compat nvidia-bumblebee

chmod +x /etc/rc.d/rc.bumblebeed
/etc/rc.d/rc.bumblebeed start

if [ -z "$(grep bumblebeed /etc/rc.d/rc.local)" ]; then
echo "if [ -x /etc/rc.d/rc.bumblebeed ]; then
  /etc/rc.d/rc.bumblebeed start
fi" >> /etc/rc.d/rc.local
fi

if [ -z "$(grep bumblebeed /etc/rc.d/rc.local_shutdown)" ]; then
echo "if [ -x /etc/rc.d/rc.bumblebeed ]; then
  /etc/rc.d/rc.bumblebeed stop
fi" >> /etc/rc.d/rc.local_shutdown
fi

## these have to be executable to work  :-)
chmod 755 /etc/rc.d/rc.local
chmod 755 /etc/rc.d/rc.local_shutdown

if [ "$UPDATE" = "yes" ]; then
  wget -N https://raw.githubusercontent.com/ryanpcmcquen/linuxTweaks/master/slackware/crazybee.sh -P ~/Bumblebee-SlackBuilds
fi

echo
echo "+===========================================+"
echo "+  Please reboot to enjoy your new driver.  +"
echo "+===========================================+"
echo



download.sh
#!/bin/sh

ARCH=$(uname -m)
PKGS=$(find -type f -name '*.info' -exec basename {} .info \;)

checksum()
{
  sum=$(md5sum $1 | cut -d ' ' -f1)

  if [ "$sum" != "$2" ]; then
    echo ""
    echo "WARNING: checksum failed: $1"
    echo ""

    sleep 2
  fi
}

for pkg in $PKGS; do
  . $pkg/$pkg.info

  if [ "$ARCH" = "x86_64" ]; then
    case $pkg in
      nvidia-bumblebee)
        DOWNLOAD=$(echo $DOWNLOAD | cut -d ' ' -f1-)
        DOWNLOAD="$DOWNLOAD $DOWNLOAD_x86_64"

        MD5SUM=$(echo $MD5SUM | cut -d ' ' -f1-)
        MD5SUM="$MD5SUM $MD5SUM_x86_64"
        ;;
      *)
        DOWNLOAD=${DOWNLOAD_x86_64:-$DOWNLOAD}
        MD5SUM=${MD5SUM_x86_64:-$MD5SUM}
    esac
  fi

  DOWNLOAD=($DOWNLOAD)
  MD5SUM=($MD5SUM)

  len=${#DOWNLOAD[@]}

  for (( i=0; i < $len; i++ )); do
    src=$(basename ${DOWNLOAD[$i]})

    echo $DOWNLOAD | grep -qi "github.com"

    if [ "$?" = "0" ]; then
      ext=$(echo $src | rev | cut -d. -f1-2 | rev)
      src=$PRGNAM-$VERSION.$ext
    fi

    if [ -e "$pkg/$src" ]; then
      if [ -f "$pkg/$src" ]; then
        checksum $pkg/$src ${MD5SUM[$i]}
      fi

      continue;
    fi

    file=$(cd $pkg; find ../ -type f -name $src)

    if [ -z "$file" ]; then
      wget -O $pkg/$src ${DOWNLOAD[$i]}
      checksum $pkg/$src ${MD5SUM[$i]}
    else
      ln -sf $file $pkg
    fi
  done
done


installer.sh
#!/bin/sh

NUMJOBS=${NUMJOBS:--j4}

export BUILD=${BUILD:-1}
export TAG=${TAG:-_SBo}
export TMP=${TMP:-/tmp/SBo}
export OUTPUT=${OUTPUT:-/tmp}
export PKGTYPE=${PKGTYPE:-tgz}

PACKAGES=( \
  libjpeg-turbo \
  VirtualGL \
  libbsd \
  bumblebee \
  bbswitch \
  nvidia-bumblebee \
  nvidia-kernel \
)

CWD=$(pwd)

if [ -z "$ARCH" ]; then
  case "$( uname -m )" in
    i?86) ARCH=i486 ;;
    arm*) ARCH=arm ;;
       *) ARCH=$( uname -m ) ;;
  esac
fi

add_rc_d()
{
  if grep -q "$2" "$1"; then
    return
  fi
  
  if ! [ -e "$1" ]; then
    echo -e "#/bin/sh\n" > $1
  fi
  
  if ! [ -x "$1" ]; then
    chmod a+x "$1"
  fi
  
  (
    if [ -n "$(tail -n 1 $1)" ]; then
      echo ""
    fi
    
    echo "if [ -x $2 ]; then"
    echo "  $2 $3"
    echo "fi"
    echo ""
  ) >> "$1"
}

del_rc_d()
{
  line=$(grep -nm 1 "$2 $3" "$1" 2>/dev/null | cut -d : -f 1)
  
  if [ -z "$line" ]; then
    return
  fi
  
  sed -i "$(($line - 1)),$(($line + 2))d" "$1"
}

download()
{
  for url in $@ ; do
    file=$(basename $url)
    path=$(find $CWD -type f -name $file | head -n 1)
    
    if [ -n "$path" ]; then
      if ! [ -e "$file" ]; then
        ln -sf "$path"
      fi
      
      continue
    fi
    
    len=$(wget --spider $url 2>&1 | grep Length | cut -d ' ' -f 2)
    
    if [ -z "$len" ]; then
      len=$(wget --spider $url 2>&1 | grep SIZE | rev | cut -d ' ' -f 1 | rev)
    fi
    
    wget -O "$file" "$url" 2>/dev/null &
    pid=""
    
    (
      while kill -0 "$pid" &>/dev/null; do
        size=$(stat -c %s $file 2>/dev/null || echo "0")
        prog=$(echo "($size / $len) * 100" | bc -l | xargs printf %1.0f)
        
        echo "XXX"
        echo "$prog"
        echo "\n$file\n\n"
        echo "Total size: $len B"
        echo "Downloaded: $size B"
        echo "XXX"
        
        sleep 1
      done
    ) |
    dialog \
      --title "Package Download" \
      --gauge "\n$file" 11 47
    
    dialog \
      --title   "Package Download" \
      --infobox "\nDownloaded: $file" 5 47
    
    sleep 2
  done
}

menu()
{
  reply=$(dialog \
    --title "Bumblebee Interactive Installer" \
    --menu  "\nWelcome to the Bumblebee interactive installer.\n\
             \nWhat would you like to do?\n" 12 52 3 \
      "Install" "Install the Bumblebee packages" \
      "Remove"  "Remove the Bumblebee packages" \
      "Exit"    "Exit the interactive installer" \
   3>&1 1>&2 2>&3)
   
   case "$reply" in
    Install)
      install
      ;;
    Remove)
      remove
      ;;
    *)
      dialog --clear
      exit
      ;;
  esac
}

install()
{
  if ! getent group bumblebee 1>/dev/null; then
    dialog \
      --title "Bumblebee Group Creation" \
      --yesno "\nA bumblebee group is required, would you like to create one?" \
        7 64
    
    if [ "$?" != "0" ]; then
      dialog \
        --title  "Bumblebee Group Creation" \
        --msgbox "\nA bumblebee group is required to proceed" 7 44
      
      menu
      return;
    fi
    
    groupadd -g 261 bumblebee
    
    if [ "$?" = "0" ]; then
      dialog \
        --title  "Bumblebee Group Creation" \
        --msgbox "\nA bumblebee group was created" 7 33
    else
      dialog \
        --title  "Bumblebee Group Creation" \
        --msgbox "\nFailed to create the bumblebee group" 7 40
    fi
  fi
  
  if [ "$ARCH" = "x86_64" ]; then
    dialog \
      --title "Multilib Support" \
      --yesno "\nThis system appears to be x86_64 based. Several of the\
               \nbumblebee dependencies can be multilib enabled allowing\
               \nfor compatability with 32-bit binaries.\n\
               \nWould you like to enable multilib support?" 11 59
    
    if [ "$?" = "0" ]; then
      export COMPAT32="yes"
    fi
  fi
  
  pkgs=()
  
  for pkg in ${PACKAGES[@]}; do
    source "$CWD/$pkg/$pkg.info"
    
    if [ -n "$(ls /var/log/packages/$pkg-* 2>/dev/null)" ]; then
      state="off"
    else
      state="on"
    fi
    
    desc=$(cat $CWD/$pkg/slack-desc | grep -m 1 "^$pkg:")
    desc=$(echo $desc | sed 's/.* (\(.*\))/\1/')
    
    pkgs+=("$pkg" "$desc" "$state")
  done
  
  reply=$(dialog --separate-output \
    --title     "Installation Selection" \
    --checklist "\nWhich packages would you like to install?" 15 67 7 \
    "${pkgs[@]}" \
   3>&1 1>&2 2>&3)
  
  if [ -z "$reply" -o "$?" != "0" ]; then
    menu
    return
  fi
  
  for pkg in $reply; do
    cd     "$CWD/$pkg"
    source "$pkg.info"
    
    if [ "$pkg" = "nvidia-bumblebee" -a "$ARCH" = "x86_64" ]; then
      download $(echo $DOWNLOAD | cut -d ' ' -f 2-)
    fi
    
    if [ -n "$DOWNLOAD_x86_64" -a "$ARCH" = "x86_64" ]; then
      download "$DOWNLOAD_x86_64"
    else
      download "$DOWNLOAD"
    fi
    
    desc=$(cat $CWD/$pkg/slack-desc | grep "^$pkg:" | sed "s/^$pkg:\s*//")
    
    dialog --cr-wrap \
      --title   "Building Package" \
      --infobox "\nStep 1 of 2: Building $pkg\n\n$desc" 16 75
    
    MAKEFLAGS="$NUMJOBS" \
    sh $pkg.SlackBuild &>/dev/null
    
    if [ "$?" != "0" ]; then
      dialog \
        --title "Building Package" \
        --yesno "\nFailed to build $pkg\n\
                 \nDo you wish to continue?" 0 0
      
      if [ "$?" != "0" ]; then
        menu
        return
      fi
    fi
    
    dialog --cr-wrap \
      --title   "Installing Package" \
      --infobox "\nStep 2 of 2: Installing $pkg\n\n$desc" 16 75
    
    upgradepkg \
      --install-new \
      --reinstall \
     $OUTPUT/$PRGNAM-$VERSION-$ARCH-$BUILD$TAG.$PKGTYPE &>/dev/null
    
    if [ "$?" != "0" ]; then
      dialog \
        --title "Installing Package" \
        --yesno "\nFailed to install $pkg\n\
                 \nDo you wish to continue?" 0 0
      
      if [ "$?" != "0" ]; then
        menu
        return
      fi
    fi
  done
  
  if echo "$reply" | grep -q bumblebee; then
    if ! grep -q rc.bumblebeed /etc/rc.d/rc.local ||
       ! grep -q rc.bumblebeed /etc/rc.d/rc.local_shutdown; then
      dialog \
        --title "Automatic Starting and Stopping" \
        --yesno "\nBumblebee requires a daemon (bumblebeed) to run in the\
                 \nbackground. For bumblebebeed to be automatically started\
                 \nand stopped with your system, the rc script (rc.bumblebeed)\
                 \nmust be added to rc.local and rc.local_shutdown.\n
                 \nWould you like to automatically start and stop bumblebeed?" \
                 11 65
      
      if [ "$?" = "0" ]; then
        add_rc_d /etc/rc.d/rc.local          /etc/rc.d/rc.bumblebeed start
        add_rc_d /etc/rc.d/rc.local_shutdown /etc/rc.d/rc.bumblebeed stop
        
        chmod a+x /etc/rc.d/rc.bumblebeed
      fi
    fi
  fi
  
  dialog \
    --title "Installation Complete" \
    --yesno "\nBumblebee and its related packages have been installed!\n\
             \nDo not forget to add users to the bumblebee group. Any\
             \nuses that need bumblebee access must be in the bumblebee\
             \ngroup. You can add users to the bumblebee group via:\n\
             \n  # usermod -G bumblebee -a USERNAME\n\n\
             \nWould you like to start the bumblebee daemon now?" 16 60
  
  if [ "$?" = "0" ]; then
    sh /etc/rc.d/rc.bumblebeed start &>/dev/null
    
    if [ "$?" = "0" ]; then
      dialog \
        --title  "Bumblebee Daemon" \
        --msgbox "\nThe bumblebee daemon has been started!\n\
                  \nYou can now test bumblebee with glxgears:\n\
                  \n  # optirun glxgears" 11 45
    else
      dialog \
        --title  "Bumblebee Daemon" \
        --msgbox "\nThe bumblebee daemon failed to start" 7 40
    fi
  fi
  
  menu
}

remove()
{
  pkgs=()
  
  for pkg in ${PACKAGES[@]}; do
    file=$(find /var/log/packages -type f -name $pkg-* | head -n 1)
    
    if [ -n "$file" ]; then
      desc=$(cat $file | grep -m 1 "^$pkg:")
      desc=$(echo $desc | sed 's/.* (\(.*\))/\1/')
      
      pkgs+=("$pkg" "$desc" "on")
    fi
  done
  
  reply=$(dialog --separate-output \
    --title     "Removal Selection" \
    --checklist "\nWhich packages would you like to remove?" 15 67 7 \
    "${pkgs[@]}" \
   3>&1 1>&2 2>&3)
  
  if [ -z "$reply" -o "$?" != "0" ]; then
    menu
    return
  fi
  
  for pkg in $reply; do
    dialog \
      --title   "Removing Package" \
      --infobox "\nRemoving $pkg" 5 0
    
    removepkg "$pkg" &>/dev/null
    
    if [ "$?" != "0" ]; then
      dialog \
      --title  "Removing Package" \
      --msgbox "\nFailed to remove $pkg" 7 0
      
      continue
    fi
  done
  
  if echo "$reply" | grep -q bumblebee; then
    del_rc_d /etc/rc.d/rc.local          /etc/rc.d/rc.bumblebeed start
    del_rc_d /etc/rc.d/rc.local_shutdown /etc/rc.d/rc.bumblebeed stop
    
    rm -f \
      /etc/bumblebee/bumblebee.conf \
      /etc/rc.d/rc.bumblebeed
    
    rmdir --ignore-fail-on-non-empty /etc/bumblebee
  fi
  
  dialog \
    --title  "Removal Complete" \
    --msgbox "\nAll selected packages, configuration, and rc scripts\
              \nhave been removed from the system" 8 56
  
  menu
}

if [ "$UID" != "0" ]; then
  dialog \
    --title  "Insufficient Permission" \
    --msgbox "\nThis script must be run as root" 7 35
  
  exit 1
fi

menu
